src/components/ScenarioParameters/UploadFileUtils.js

```diff
@@ -3,13 +3,38 @@
 
 import React from 'react';
 import { UPLOAD_FILE_STATUS_KEY } from '@cosmotech/ui';
-import { ORGANIZATION_ID, WORKSPACE_ID } from '../../config/AppInstance';
+import {
+  ORGANIZATION_ID, WORKSPACE_ID,
+  ENABLE_APPLICATION_INSIGHTS
+} from '../../config/AppInstance';
 import DatasetService from '../../services/dataset/DatasetService';
 import WorkspaceService from '../../services/workspace/WorkspaceService';
 import { FileUpload } from './components/tabs';
+import { AppInsights } from '../../services/AppInsights';
 
 export const STORAGE_ROOT_DIR_PLACEHOLDER = '%WORKSPACE_FILE%/';
 
+// App Insigths
+const appInsights =
+  ENABLE_APPLICATION_INSIGHTS
+    ? AppInsights.getInstance()
+    : undefined;
```

Comment:
Can we factorize the `ENABLE_APPLICATION_INSIGHTS` parameter check by putting it inside the `AppInsights` service ? If `ENABLE_APPLICATION_INSIGHTS` is false, then we could either return null or undefined when getting the singleton, or even better, prevent the service from monitoring events and metrics.

This would allow us to remove all the `if (appInsights) {...}` checks in the views and components